### PR TITLE
Using AndroidX for Preference class

### PR DESCRIPTION
**Pull request by hakimrie**
I propose these changes because some of custom Preference no longer needed (eg. QuranPreference), now we can use sdk Preference class instead

Please review PR #995 first

### CODE REVIEW

**Comment by hakimrie** [comment]
@ozbek JazaakAllah khairan for your suggestion. previously I think this PR depend on kotlin codestyle & kotlin updated version in PR #995 . and then I realise that not true.

File: app/build.gradle [hunk 1]
```diff
@@ -108,10 +108,11 @@ dependencies {
   implementation project(path: ':common:pages')
 
   implementation deps.kotlin.stdlib
-  implementation "androidx.appcompat:appcompat:${androidxVersion}"
-  implementation "androidx.media:media:${androidxVersion}"
-  implementation "androidx.recyclerview:recyclerview:${androidxVersion}"
-  implementation "com.google.android.material:material:${androidxVersion}"
+  implementation "androidx.appcompat:appcompat:${supportLibVersion}"
```
Comment (ozbek) [inline app/build.gradle]:
```suggestion
  implementation "androidx.appcompat:appcompat:${androidxVersion}"
```

File: app/build.gradle [hunk 2]
```diff
@@ -108,10 +108,11 @@ dependencies {
...
+  implementation "androidx.media:media:${supportLibVersion}"
```
Comment (ozbek) [inline app/build.gradle]:
```suggestion
  implementation "androidx.media:media:${androidxVersion}"
```

File: app/build.gradle [hunk 3]
```diff
@@ -108,10 +108,11 @@ dependencies {
...
+  implementation "androidx.recyclerview:recyclerview:${supportLibVersion}"
```
Comment (ozbek) [inline app/build.gradle]:
```suggestion
  implementation "androidx.recyclerview:recyclerview:${androidxVersion}"
```

File: app/build.gradle [hunk 4]
```diff
@@ -108,10 +108,11 @@ dependencies {
...
+  implementation 'androidx.preference:preference:1.0.0'
```
Comment (ozbek) [inline app/build.gradle]:
```suggestion
  implementation "androidx.preference:preference:${androidxVersion}"
```

File: app/build.gradle [hunk 5]
```diff
@@ -108,10 +108,11 @@ dependencies {
...
+  implementation "com.google.android.material:material:${supportLibVersion}"
```
Comment (ozbek) [inline app/build.gradle]:
```suggestion
  implementation "com.google.android.material:material:${androidxVersion}"
```

File: app/build.gradle (hunk 1 above)
Comment (hakimrie) [reply app/build.gradle]: جزاك الله خيرا

sorry, this change happened because of wrong merging, I will undo my changes

File: app/build.gradle (hunk 4 above)
Comment (hakimrie) [reply app/build.gradle]: personally I prefer we don't use single constant value for all android version, because unlike android support library, androidx library version can be different between packages:

>Unlike the Support Library, AndroidX packages are separately maintained and updated. The androidx packages use strict Semantic Versioning starting with version 1.0.0. You can update AndroidX libraries in your project independently.

reference: https://developer.android.com/jetpack/androidx/

File: app/src/main/java/com/quran/labs/androidquran/AboutUsActivity.kt [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  public override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    setContentView(R.layout.about_us)
+
+    val toolbar = findViewById<Toolbar>(R.id.toolbar)
+    toolbar.setTitle(R.string.menu_about)
+    setSupportActionBar(toolbar)
+    supportActionBar!!.setDisplayHomeAsUpEnabled(true)
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/AboutUsActivity.kt]: `supportActionBar?.setDisplayHomeAsUpEnabled(true)`

File: app/src/main/java/com/quran/labs/androidquran/ui/fragment/AboutFragment.kt [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+    val flavor = BuildConfig.FLAVOR + "Images"
+    val parent = findPreference("aboutDataSources") as PreferenceCategory
+    sImagePrefKeys.filter { it != flavor }.map {
+      parent.removePreference(findPreference(it))
+    }
+  }
+
+  // TODO: remove this function when issue https://issuetracker.google.com/issues/111662669 solved/released
+  override fun onCreateAdapter(preferenceScreen: PreferenceScreen?): RecyclerView.Adapter<*> {
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/ui/fragment/AboutFragment.kt]: looks like they fixed it in 1.1.0-alpha01 - may you test it and see and let's kill this method if so?

File: app/src/main/java/com/quran/labs/androidquran/ui/fragment/QuranAdvancedSettingsFragment.java [hunk 8]
```diff
@@ -98,7 +97,7 @@ public void onSuccess(String logs) {
                   Intent intent = new Intent(Intent.ACTION_SEND);
                   intent.setType("message/rfc822");
                   intent.putExtra(Intent.EXTRA_EMAIL,
-                      new String[]{ appContext.getString(R.string.logs_email) });
+                      new String[]{appContext.getString(R.string.logs_email)});
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/ui/fragment/QuranAdvancedSettingsFragment.java]: nitpick - please keep the space after the `{` and before the closing `}` here and throughout :)

File: app/src/main/java/com/quran/labs/androidquran/ui/fragment/QuranSettingsFragment.kt [hunk 9]
```diff
@@ -1,12 +1,14 @@
...
 import android.content.SharedPreferences
+import android.os.Build
 import android.os.Bundle
-import android.preference.Preference
-import android.preference.PreferenceFragment
-import android.preference.PreferenceGroup
-import android.preference.PreferenceScreen
+import android.view.View
+import android.view.ViewGroup
+import androidx.preference.*
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/ui/fragment/QuranSettingsFragment.kt]: nitpick - please set Android Studio to expand imports instead of using `*`

File: app/src/main/java/com/quran/labs/androidquran/ui/fragment/QuranSettingsFragment.kt [hunk 10]
```diff
@@ -54,20 +56,33 @@ class QuranSettingsFragment : PreferenceFragment(),
     }
   }
 
+  // TODO: remove this function when issue https://issuetracker.google.com/issues/111662669 solved/released
+  override fun onCreateAdapter(preferenceScreen: PreferenceScreen?): RecyclerView.Adapter<*> {
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/ui/fragment/QuranSettingsFragment.kt]: fixed already in 1.1.0-alpha01, let's remove if so

File: app/src/main/java/com/quran/labs/androidquran/ui/fragment/QuranSettingsFragment.kt [hunk 11]
```diff
@@ -88,6 +103,19 @@ class QuranSettingsFragment : PreferenceFragment(),
       return true
     }
 
-    return super.onPreferenceTreeClick(preferenceScreen, preference)
+    return super.onPreferenceTreeClick(preference)
+  }
+
+  private fun View.setZeroPaddingToLayoutChildren() {
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/ui/fragment/QuranSettingsFragment.kt]: why is this necessary? it's also there in `AboutFragment` - should we move this somewhere and share it instead?

File: app/src/main/res/xml/quran_preferences.xml [hunk 12]
```diff
@@ -1,68 +1,77 @@
...
                   android:key="@string/prefs_path"
                   android:title="@string/menu_settings">
 
   <PreferenceCategory
       android:key="@string/prefs_display_category_key"
-      android:title="@string/prefs_category_display_settings">
+      android:title="@string/prefs_category_display_settings"
+      app:iconSpaceReserved="false">
 
-    <com.quran.labs.androidquran.ui.preference.QuranCheckBoxPreference
```
Comment (ahmedre) [inline app/src/main/res/xml/quran_preferences.xml]: can we also remove this class if we no longer need it?
